source/utils/logging.cpp

```diff
@@ -7,4 +7,87 @@
  */
 
 #include "logging.h"
-namespace loader {} // namespace loader
\ No newline at end of file
+namespace loader {
+
+std::string to_string(const ze_result_t result) {
```

Comment:
Is there a way we can stringify this with MACROs as opposed to keeping another list we have to maintain? Else we have to modify two locations with potential mismatch with new ERROR codes introduced.

Reply:
What is the other place where we have a list like this?

I'm not sure if macro would help much but we could probably generate this funtion from a .mako file.

Anyway, I would prefer to do that in a separate PR, I did not introduce this function - I just moved it from ze_loader.cpp

Reply:
Lets take this proposal to a future PR.